cogs/tempchannels/tempchannels.py

```diff
@@ -536,37 +541,42 @@ async def tempChannelsDenyRemove(self, ctx: Context, *, role: discord.Role):
     async def tempChannelsDelete(self, ctx: Context):
         """Deletes the temp channel, if it exists."""
-        async with self.config.guild(ctx.guild).all() as guildData:
-            if guildData[KEY_CH_CREATED] and guildData[KEY_CH_ID]:
-                # Channel created, see when we should delete it.
-                try:
-                    chanObj = self.bot.get_channel(guildData[KEY_CH_ID])
-                    await chanObj.delete()
-                except discord.DiscordException:
-                    self.logger.error("Could not delete channel!", exc_info=True)
-                    await ctx.send(
-                        ":warning: TempChannel: Something went wrong "
-                        "while trying to delete the channel. Please "
-                        "check the console log for details."
-                    )
-                else:
-                    guildData[KEY_CH_ID] = None
-                    guildData[KEY_CH_CREATED] = False
-                    self.logger.info(
-                        "%s (%s) deleted the temp channel #%s (%s) in %s (%s).",
-                        ctx.author.name,
-                        ctx.author.id,
-                        chanObj.name,
-                        chanObj.id,
-                        ctx.guild.name,
-                        ctx.guild.id,
-                    )
-                    await ctx.send(":white_check_mark: TempChannel: Channel deleted")
-            else:
+        guildConfig = self.config.guild(ctx.guild)
+        channelId = await guildConfig.get_attr(KEY_CH_ID)()
+        channelCreated = await guildConfig.get_attr(KEY_CH_CREATED)()
+
+        if channelCreated and channelId:
+            # Channel created, see when we should delete it.
+            try:
+                chanObj = self.bot.get_channel(channelId)
+                await chanObj.delete()
+            except discord.DiscordException:
+                self.logger.error("Could not delete channel!", exc_info=True)
                 await ctx.send(
-                    ":negative_squared_cross_mark: TempChannel: There is no "
-                    "temporary channel to delete!"
+                    ":warning: TempChannel: Something went wrong "
+                    "while trying to delete the channel. Please "
+                    "check the console log for details."
                 )
+            else:
+                channelId = None
+                channelCreated = False
+                self.logger.info(
+                    "%s (%s) deleted the temp channel #%s (%s) in %s (%s).",
+                    ctx.author.name,
+                    ctx.author.id,
+                    chanObj.name,
+                    chanObj.id,
+                    ctx.guild.name,
+                    ctx.guild.id,
+                )
+                await guildConfig.get_attr(KEY_CH_ID).set(channelId)
+                await guildConfig.get_attr(KEY_CH_CREATED).set(channelCreated)
```

Comment:
any particular reason we can't set `None` and `False` directly?

Reply:
I do that to preserve the old code flow. Other than that I can see the redundancy.
The next commit should address this.